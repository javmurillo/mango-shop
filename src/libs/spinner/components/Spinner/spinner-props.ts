export interface CustomSpinnerProps {
  error: boolean;
  message: string;
}
