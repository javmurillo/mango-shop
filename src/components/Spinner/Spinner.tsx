import { Spinner } from 'react-bootstrap';
import styled from 'styled-components';
import { CustomSpinnerProps } from './spinner-props';

const StyledSpinnerWrapper = styled.div`
  display: flex;
  justify-content: center;
`;

const StyledSpinner = styled(Spinner)`
  text-align: center;
`;

export const CustomSpinner = (props: CustomSpinnerProps): JSX.Element => {
  return props.error ? (
    <p>{props.message}</p>
  ) : (
    <StyledSpinnerWrapper>
      <StyledSpinner animation="border" role="status">
        <span className="sr-only">Loading...</span>
      </StyledSpinner>
    </StyledSpinnerWrapper>
  );
};
