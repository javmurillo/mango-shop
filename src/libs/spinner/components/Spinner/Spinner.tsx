import { Spinner } from 'react-bootstrap';
import styled from 'styled-components';
import { CustomSpinnerProps } from './spinner-props';

const StyledSpinnerWrapper = styled.div`
  display: flex;
  justify-content: center;
`;

const StyledSpinner = styled(Spinner)`
  text-align: center;
`;

/**
 * Custom spinner which shows an error message in case of [error] equals true. Icon spinning otherwise.
 * @param props CustomSpinnerProps
 */
export const CustomSpinner = (props: CustomSpinnerProps): JSX.Element => {
  return props.error ? (
    <div aria-label="spinner">
      <p aria-label="error message">{props.message}</p>
    </div>
  ) : (
    <StyledSpinnerWrapper aria-label="spinner">
      <StyledSpinner animation="border" role="status">
        <span className="sr-only">Loading...</span>
      </StyledSpinner>
    </StyledSpinnerWrapper>
  );
};
