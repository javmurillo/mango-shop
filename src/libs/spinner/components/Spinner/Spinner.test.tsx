import { cleanup, render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import { CustomSpinner } from './Spinner';

afterEach(cleanup);

const setup = (error: boolean) => {
  const utils = render(<CustomSpinner message="Test error" error={error} />);
  const spinner = utils.getByLabelText('spinner');
  return {
    spinner,
    ...utils,
  };
};

describe('<NotFound/> tests', () => {
  test('renders correctly with error', () => {
    const { spinner } = setup(true);
    expect(spinner).toBeInTheDocument();
  });

  test('renders correctly with no error', () => {
    const { spinner } = setup(false);
    expect(spinner).toBeInTheDocument();
  });

  test('shoud match snapshot', () => {
    const testRendererJsonError = TestRenderer.create(
      <CustomSpinner message="Test error" error={true} />
    ).toJSON();
    expect(testRendererJsonError).toMatchSnapshot();
    const testRendererJsonNoError = TestRenderer.create(
      <CustomSpinner message="Test error" error={false} />
    ).toJSON();
    expect(testRendererJsonNoError).toMatchSnapshot();
  });
});
