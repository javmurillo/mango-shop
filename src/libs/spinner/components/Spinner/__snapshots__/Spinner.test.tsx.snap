// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound/> tests shoud match snapshot 1`] = `
<div
  aria-label="spinner"
>
  <p
    aria-label="error message"
  >
    Test error
  </p>
</div>
`;

exports[`<NotFound/> tests shoud match snapshot 2`] = `
<div
  aria-label="spinner"
  className="sc-bdfBwQ dHJFwu"
>
  <div
    className="sc-gsTCUz bGKjuf spinner-border"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
